Software/controller/controller.py

```diff
@@ -82,90 +83,97 @@ def on_message(client, userdata, msg):
-                            break
-                        if placeInList == self.model.list_of_places[-1]:
-                            # Found no vacant place - put number in queue
-                            print("New number " + str(new_number))
-                            self.model.list_of_ticket_numbers.append(new_number)
-                            self.update_queue()
+                    handle_message_from_controller(self, json_loaded)
             except Exception as e:
                 print(str(e))
                 print("Something went wrong on mqtt reception")
+            self.update_queue(self.model.list_of_processing_times, self.model.list_of_labels_to_display_in_queue,
+                              self.model.list_of_ticket_numbers)
+
+        def handle_message_from_controller(self, json_loaded):
+            new_number = json_loaded["new_number"]
+            print("Received new ticket number from controller: " +
+                  str(new_number))
+            if new_number < 0:
+                raise Exception('New ticket number must be positive!')
+            # Check if new number is already in use
+            for placeInList in self.model.list_of_places:
+                if placeInList.ticket_number == new_number:
+                    print("New number " + str(new_number) +
+                          " already registered")
+                    return
+            if new_number in self.model.list_of_ticket_numbers:
+                print("New number " + str(new_number) +
+                      "already in ticket number list")
+                return
+            for count, placeInList in enumerate(self.model.list_of_places):
+                # Search an empty place
+                if placeInList.is_free():
+                    # Found one - register ticket number to place
+                    print("Register ticket: " +
+                          str(new_number) + " to free place " + str(count + 1))
+                    reservePlaceForTicketNumber(self, count, new_number)
+                    break
+                if placeInList == self.model.list_of_places[-1]:
+                    # Found no vacant place - put number in queue
+                    print("New number " + str(new_number))
+                    self.model.list_of_ticket_numbers.append(new_number)
+
+        def handle_message_from_place(self, json_loaded):
+            if type(json_loaded["place_occupied"]) != bool:
+                raise Exception('Key "place_occupied" is of wrong type! Must be boolean!')
+            place_number = json_loaded["place_number"] - 1
+            if place_number > numberOfPlaces or place_number < 0:
+                raise Exception('Received place_number ' + str(place_number) + ' refers to not existing place!')
+            if json_loaded["place_occupied"] == True:
+                # Place was taken by the owner of the ticket_number
+                # Place number in MQTT-Message starts with 1 and must be decremented
```

Comment:
I guess the `# Place number in MQTT-Message starts with 1 and must be decremented` comment belongs to `place_number =` some lines above, please move it there.

Reply:
nice catch!